src/features/movies/moviesSaga.js

```diff
@@ -20,8 +20,10 @@ function* fetchGenresHandler() {
 function* fetchMoviesHandler({ payload }) {
   try {
     const currentPage = payload.currentPage;
-    const data = yield call(getPopularData, "movie", currentPage);
-    yield delay(500);
+    const query = payload.query;
+    const data = yield !query
+      ? call(getPopularData, "movie", currentPage)
+      : call(getDataByQuery, "movie", currentPage, query);
```

Comment:
I suggest to not start with negation. Let's remove the `!` and swap these two calls 
`    const data = yield query
      ? call(getDataByQuery, "movie", currentPage, query)
      : call(getPopularData, "movie", currentPage);`

Reply:
Done.